### PR TITLE
feat(topology): wire up node search + viewport/layout/routing fixes

### PR DESCRIPTION
**Pull request by nadaverell**
## Summary

Revives the previously-orphaned `TopologySearch` and wires it into the topology view, then fixes a cluster of layout, viewport, and navigation issues surfaced while testing it.

**Topology node search.** `TopologySearch` was exported from `@skyhook-io/k8s-ui` but mounted by no consumer (born unwired, carried into the package by the original extraction). It's now rendered in the topology view — `/` or the on-canvas button opens it; picking a result selects the node and pans/zooms the canvas to it. When the target lives inside a collapsed group, the group expands and the viewport fits to it.

**Selected-node highlight.** Clicking a node (drawer open) now actually highlights it on the canvas. The previous wiring never matched — it rebuilt a `kind/ns/name` id with the wrong separators; we now look the real node id up from the topology. The selected node gets a prominent brand outline + glow and is lifted above its siblings, and the highlight re-applies after a relayout (e.g. search-driven expand).

**Controls + viewport.** The bottom-left controls are split into two grouped pills (viewport: zoom/fit/export/pause; level-of-detail: collapse/cards/expand) instead of one long 8-icon strip, with clearer icons and no duplicate "minus". Search no longer overlaps the namespace breadcrumb. Bulk level changes, per-group level changes, and grouping-mode changes now re-fit the viewport once the async ELK relayout settles (debounced), so e.g. "expand all" actually frames the expanded graph.

**Layout overflow.** Node dimensions were declared 56px tall while cards render ~76–79px, so cards overflowed their group boundaries. Card heights are bumped to 84 so groups size correctly.

**Routing (`web/src/App.tsx`).** Fixes a pre-existing blank-page-on-back crash (React #185 "maximum update depth"): the URL-write and URL-read effects form a two-way binding that oscillated on a history Pop (back/forward) because the writer ran with stale state and reverted the popped URL. The writer now suppresses its reverting writes during the synchronous post-Pop reconciliation burst (the guard auto-clears on the next macrotask, so later user-driven writes are unaffected) — the popped URL is authoritative.

## Notes
- The routing fix is in `App.tsx` (radar-app), separate in nature from the k8s-ui topology changes; it just surfaced via the topology → full-view → back flow.
- Follow-up worth tracking separately: retire the two-way URL↔state binding in favour of a single source of truth (`useSearchParams`) — the current fix is a correct targeted patch, not a redesign.
- Verified throughout with `/visual-test` against the gitops-demo cluster; `make tsc` clean; topology unit tests pass.

---

> [!NOTE]
> **Medium Risk**
> App routing guard touches global URL sync on history Pop; topology selection/viewport logic runs on async ELK relayout and could regress edge cases, but changes are scoped to topology UX.
> 
> **Overview**
> **Topology search and canvas focus.** `TopologySearch` is mounted on the topology view (`/` opens it). Choosing a result opens the resource and drives `focusNodeId` / `focusNonce` on `TopologyGraph` to pan/zoom; collapsed groups expand first, then the viewport fits the group after ELK settles.
> 
> **Selection and layout.** Canvas highlight now uses real topology node ids (lookup in `App.tsx`) instead of a reconstructed id. Selected nodes get `.topology-node-selected` styling, higher `zIndex`, and selection re-applies after relayout. ELK node heights move from 56px to 84px so cards no longer overflow groups.
> 
> **Viewport and controls.** Bottom-left controls split into two pills (LOD vs zoom/fit/export/pause) with updated icons. Bulk/grouping level changes and group expand/collapse trigger debounced `fitView` after relayout. `TopologySearch` gains configurable trigger placement and shows `/` as the shortcut.
> 
> **Routing.** `App.tsx` suppresses query-param URL writes briefly after browser back/forward to stop URL↔state oscillation (React #185 blank page).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4c0108b7374fbe157ebb97406111bb8382e20eaf. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: web/src/App.tsx [hunk 1]
```diff
@@ -843,10 +827,28 @@ function AppInner() {
...
+  // re-render burst, so a macrotask-deferred clear covers it; clearing
+  // afterward means a stale arm can't survive into an unrelated later write
+  // (e.g. a Pop that changes none of the write effect's deps would otherwise
+  // leave the flag set and silently drop the next user-driven URL write).
+  useEffect(() => {
+    if (navigationType !== NavigationType.Pop) return
+    skipUrlWriteAfterPopRef.current = true
+    const id = setTimeout(() => { skipUrlWriteAfterPopRef.current = false }, 0)
+    return () => clearTimeout(id)
+  }, [location.key, navigationType])
```
[the remark this answers is not in the document]
Comment (nadaverell) [reply web/src/App.tsx:848]: Good catch — fixed in 4c0108b. The guard now resets on any non-Pop navigation, so a Push/Replace landing before the macrotask clear can't leave it stuck. Together with the macrotask clear (which covers the Pop where the write effect doesn't run), both stuck-flag paths are closed.
